servers/src/main/java/tachyon/master/lineage/LineageMaster.java

```diff
@@ -0,0 +1,415 @@
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.master.lineage;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+
+import org.apache.thrift.TProcessor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import com.google.common.collect.Sets;
+
+import tachyon.Constants;
+import tachyon.HeartbeatThread;
+import tachyon.TachyonURI;
+import tachyon.client.file.TachyonFile;
+import tachyon.conf.TachyonConf;
+import tachyon.exception.ExceptionMessage;
+import tachyon.job.Job;
+import tachyon.master.MasterBase;
+import tachyon.master.MasterContext;
+import tachyon.master.file.FileSystemMaster;
+import tachyon.master.journal.Journal;
+import tachyon.master.journal.JournalEntry;
+import tachyon.master.journal.JournalOutputStream;
+import tachyon.master.lineage.checkpoint.CheckpointPlan;
+import tachyon.master.lineage.checkpoint.CheckpointPlanningExecutor;
+import tachyon.master.lineage.journal.AsyncCompleteFileEntry;
+import tachyon.master.lineage.journal.LineageEntry;
+import tachyon.master.lineage.journal.LineageIdGeneratorEntry;
+import tachyon.master.lineage.journal.PersistFilesEntry;
+import tachyon.master.lineage.journal.RequestFilePersistenceEntry;
+import tachyon.master.lineage.meta.Lineage;
+import tachyon.master.lineage.meta.LineageFile;
+import tachyon.master.lineage.meta.LineageFileState;
+import tachyon.master.lineage.meta.LineageIdGenerator;
+import tachyon.master.lineage.meta.LineageStore;
+import tachyon.master.lineage.meta.LineageStoreView;
+import tachyon.master.lineage.recompute.RecomputeExecutor;
+import tachyon.master.lineage.recompute.RecomputePlanner;
+import tachyon.thrift.BlockInfoException;
+import tachyon.thrift.BlockLocation;
+import tachyon.thrift.CheckpointFile;
+import tachyon.thrift.CommandType;
+import tachyon.thrift.FileAlreadyExistException;
+import tachyon.thrift.FileBlockInfo;
+import tachyon.thrift.FileDoesNotExistException;
+import tachyon.thrift.InvalidPathException;
+import tachyon.thrift.LineageCommand;
+import tachyon.thrift.LineageDeletionException;
+import tachyon.thrift.LineageDoesNotExistException;
+import tachyon.thrift.LineageInfo;
+import tachyon.thrift.LineageMasterService;
+import tachyon.util.ThreadFactoryUtils;
+import tachyon.util.io.PathUtils;
+
+/**
+ * The lineage master stores the lineage metadata in Tachyon, and it contains the components that
+ * manage all lineage-related activities.
+ */
+public final class LineageMaster extends MasterBase {
```

Comment:
documentation for this class is too thin for me and prevents me understanding it deeper. Please at least javadoc all public methods in this class.